src/App.js

```diff
@@ -14,11 +16,58 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    selectedGood: '',
+  };
+
+  selectedGoodHandler = (good) => {
+    this.setState({
+      selectedGood: good,
+    });
+  };
+
+  clearGoodsListHandler = () => {
+    this.setState({
+      selectedGood: '',
+    });
+  }
+
+  render() {
+    const { selectedGood } = this.state;
+
+    return (
+      <div className="App">
+        {
+          selectedGood && (
+            <button
+              onClick={this.clearGoodsListHandler}
+              className="clear"
+              type="button"
+            >
+              Clear
+            </button>
+          )
+        }
+        <h1>{`Selected good: ${selectedGood}`}</h1>
+
+        {goodsFromServer.map(good => (
+          <button
+            key={good}
+            onClick={e => this.selectedGoodHandler(good)}
```

Comment:
```suggestion
            onClick={() => this.selectedGoodHandler(good)}
```
you don't use event here